### PR TITLE
fix: export storage types bug

### PR DESCRIPTION
**Pull request by goldbez**
#### Description of changes
The function for generating types was getting called when the cfn files were not yet generated for all resources. This happened because the order of the array of resources did not have custom as the last resource. The types were being built in the custom category. The fix is to build the types again after all of the cfn files have been created in amplify init. 

#### Issue #, if available
[# 11847](https://github.com/aws-amplify/amplify-cli/issues/11847)

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by akshbhu — APPROVED** [review]

File: packages/amplify-e2e-tests/custom-resources/custom-cdk-stack-with-storage.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    // eslint-disable-next-line no-new
+    new cdk.CfnParameter(this, 'env', {
+      type: 'String',
+      description: 'Current Amplify CLI env name',
+    });
+
+    /* AWS CDK Code goes here - Learn more: https://docs.aws.amazon.com/cdk/latest/guide/home.html */
+
+    /* Example 2: Adding IAM role to the custom stack */
+    const role = new iam.Role(this, 'CustomRole', {
```
[the remark this answers is not in the document]
Comment (jhockett) [reply packages/amplify-e2e-tests/custom-resources/custom-cdk-stack-with-storage.ts:25]: No need to assign a variable if not referencing it.

File: packages/amplify-e2e-tests/custom-resources/custom-cdk-stack-with-storage.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    /* Example 3: Adding policy to the IAM role*/
+    role.addToPolicy(
+      new iam.PolicyStatement({
+        actions: ['*'],
+        resources: [topic.topicArn],
+      }),
+    );
+
+    const amplifyResources: AmplifyDependentResourcesAttributes = AmplifyHelpers.addResourceDependency(
```
[the remark this answers is not in the document]
Comment (jhockett) [reply packages/amplify-e2e-tests/custom-resources/custom-cdk-stack-with-storage.ts:30]: Same here

**Review by jhockett — COMMENTED** [review]
@goldbez I see that the build is failing in the CI checks and a failing e2e, can you check those?

**Review by jhockett — DISMISSED** [review]

File: packages/amplify-e2e-core/src/init/initProjectHelper.ts [hunk 3]
```diff
@@ -492,3 +492,15 @@ export function amplifyStatus(cwd: string, expectedStatus: string, testingWithLa
...
+export function initHeadless(cwd: string, envName: string, appId: string): Promise<void> {
+  return new Promise((resolve, reject) => {
+    spawn(getCLIPath(), ['init', '--yes', '--envName', envName, '--appId', appId], { cwd, stripColors: true }).run((err: Error) => {
+      if (!err) {
+        resolve();
+      } else {
+        reject(err);
+      }
+    });
+  });
```
Comment (sobolk) [inline packages/amplify-e2e-core/src/init/initProjectHelper.ts:505]:
```suggestion
  return spawn(getCLIPath(), ['init', '--yes', '--envName', envName, '--appId', appId], { cwd, stripColors: true }).runAsync();
```

File: packages/amplify-e2e-core/src/utils/git-operations.ts [hunk 4]
```diff
@@ -24,6 +24,11 @@ export const gitCleanFdx = async (cwd: string): Promise<void> => {
   await execa('git', ['clean', '-fdx'], { cwd });
 };
 
+// It is important to note the difference between -fdx and -fdX
```
Comment (sobolk) [inline packages/amplify-e2e-core/src/utils/git-operations.ts:27]: can you add a sentence that explains the difference ?

**Review by sobolk — APPROVED** [review]

**Review by jhockett — APPROVED** [review]

**Pull request merged**